spec/lib/app_helpers/taxon_breadcrumbs_spec.rb

```diff
@@ -0,0 +1,154 @@
+require 'spec_helper'
+
+RSpec.describe GovukPublishingComponents::AppHelpers::TaxonBreadcrumbs do
+  describe "Taxon breadcrumbs" do
+    it "can handle any valid content item" do
+      payload = GovukSchemas::RandomExample.for_schema(
+        frontend_schema: "taxon",
+      )
+
+      expect {
+        described_class.new(payload).breadcrumbs
+      }.to_not raise_error
+    end
+
+    it "returns the root when taxon is not specified" do
+      content_item = {
+        "title" => "Some Content",
+        "document_type" => "answer",
+        "links" => {},
+      }
+      breadcrumbs = breadcrumbs_for(content_item)
+
+      expect(breadcrumbs).to eq(
+        [
+          { title: "Home", url: "/", is_page_parent: true },
+        ]
+      )
+    end
+
+    it "places parent under the root when it's a root-level taxon" do
+      content_item = taxon_with_parent_taxons([])
+
+      breadcrumbs = breadcrumbs_for(content_item)
+
+      expect(breadcrumbs).to eq(
+        [
+          { title: "Home", url: "/", is_page_parent: true },
+        ]
+      )
+    end
+
+    context 'with a taxon with taxon parents' do
+      it "includes parents and grandparents when available" do
+        grandparent = {
+          "title" => "Another-parent",
+          "base_path" => "/another-parent",
+          "content_id" => "30c1b93d-2553-47c9-bc3c-fc5b513ecc32",
+          "locale" => "en",
+          "phase" => "live",
+        }
+
+        parent = {
+          "content_id" => "30c1b93d-2553-47c9-bc3c-fc5b513ecc32",
+          "locale" => "en",
+          "title" => "A-parent",
+          "base_path" => "/a-parent",
+          "phase" => "live",
+          "links" => {
+            "parent_taxons" => [grandparent]
+          }
+        }
+
+        content_item = taxon_with_parent_taxons([parent])
+        breadcrumbs = breadcrumbs_for(content_item)
+
+        expect(breadcrumbs).to eq(
+          [
+            { title: "Home", url: "/", is_page_parent: false },
+            { title: "Another-parent", url: "/another-parent", is_page_parent: false },
+            { title: "A-parent", url: "/a-parent", is_page_parent: true },
+          ]
+        )
+      end
+    end
+
+    context 'with a taxon content item with parent taxons' do
+      it "includes parents and grandparents when available" do
```

Comment:
I'd rename this, as grandparent taxons aren't being tested.

Reply:
Removed!